src/utils/ManifestLoader.ts

```diff
@@ -1,19 +1,71 @@
 import manifests from "@src/data/manifests-latest.json";
+import { fromZodError } from "zod-validation-error";
+import { z, ZodError } from "zod";
 
-type NodeElement = {
-  name: string;
-  type: string;
-  key: string;
-  inputs?: { name: string; id: string; type: string }[];
-  ui_component_id?: string;
-  pip_dependencies?: Array<{
-    name: string;
-    v?: string | number;
-  }>;
-}[];
+const commandsSchema = z.array(
+  z.object({
+    name: z.string(),
+    key: z.string(),
+    type: z.string(),
+    inputs: z.optional(
+      z.array(z.object({ name: z.string(), id: z.string(), type: z.string() }))
+    ),
+    ui_component_id: z.optional(z.string()),
+    pip_dependencies: z.optional(
+      z.array(z.object({ name: z.string(), v: z.optional(z.string()) }))
+    ),
+  })
+);
+
+const paramsSchema = z.record(
+  z.string(),
+  z.record(
+    z.string(),
+    z.object({
+      type: z.string(),
+      default: z.union([z.string(), z.number(), z.boolean()]),
+      options: z.optional(z.array(z.string())),
+    })
+  )
+);
+
+const manifestSchema = z.object({
+  commands: commandsSchema,
+  parameters: paramsSchema,
+});
+
+export type Manifest = z.infer<typeof manifestSchema>;
+export type ManifestParams = z.infer<typeof paramsSchema>;
+export type ManifestCommands = z.infer<typeof commandsSchema>;
+
+export function getManifestParams() {
+  try {
+    const parsedManifest: Manifest = manifestSchema.parse(manifests);
+    return parsedManifest.parameters;
+  } catch (e) {
+    if (e instanceof ZodError) {
+      throw fromZodError(e);
+    } else {
+      throw new Error("something is seriously wrong");
```

Comment:
Should probably throw a more descriptive error message (just in case this actually happens, then we would have a better idea of how to debug it)